native_words.asm

```diff
@@ -5843,39 +5818,34 @@ xt_plus_loop:
                 lda cp
                 sec
                 sbc #1
-                sta tmp2
+                pha             ; lsb
                 lda cp+1
                 sbc #0
-                sta tmp2+1
+                pha             ; msb
 
-                ; now compile this in the DO/?DO routine
+                ; now compile our template in the DO/?DO routine
                 ldy #0
-
-                lda #$A9        ; opcode for LDA immediate
-                sta (tmp1),y
-                iny
-                lda tmp2+1      ; MSB
-                sta (tmp1),y
-                iny
-                lda #$48        ; Opcode for PHA
-                sta (tmp1),y
-                iny
-
-                lda #$A9        ; opcode for LDA immediate
-                sta (tmp1),y
-                iny
-                lda tmp2        ; LSB
+-
+                lda loop_epilogue,y
+                bne +
+                pla
++
                 sta (tmp1),y
                 iny
-                lda #$48        ; Opcode for PHA
-                sta (tmp1),y
+                cpy #6
```

Comment:
If you put a label at the end of the epilogue, you can use label math here instead of a magic number.  That might also be handy if you end up fundamentally changing how the loops work and need to edit the epilogue.